client/state/posts/reducer.js

```diff
@@ -282,6 +281,23 @@ export function edits( state = {}, action ) {
 				[ action.siteId ]: omit( state[ action.siteId ], action.postId || '' )
 			} );
 
+		case POST_SAVE_SUCCESS:
+			if ( ! state.hasOwnProperty( action.siteId ) || ! action.savedPost ) {
+				break;
+			}
+			const { postId, siteId, savedPost } = action;
+
+			if ( postId ) {
+				return state;
+			}
+
+			// if postId is null, copy over any edits
+			return Object.assign( {}, state, {
+				[ siteId ]: {
```

Comment:
The way we're assigning here will delete any other post IDs for the same site. While in practice this will probably never occur, and conveniently removes the `''` (empty) key for us, we might want to preserve these either with `_.merge` or a nested `Object.assign`:

```js
return {
	...state,
	[ siteId ]: {
		...state[ siteId ],
		[ savedPost.ID ]: state[ siteId ][ '' ]
	}
};
```

Of course at that point we'd need to assume responsibility for removing the empty key:

```js
const nextState = {
	...state,
	[ siteId ]: {
		...state[ siteId ],
		[ savedPost.ID ]: state[ siteId ][ '' ]
	}
};
delete nextState[ siteId ][ '' ];
return nextState;
```

Alternatively we could use [`_.mapKeys`](https://lodash.com/docs/4.17.4#mapKeys) to reassign the empty key:

```js
return {
	...state,
	[ siteId ]: mapKeys( state[ siteId ], ( value, key ) => (
		'' === key ? savedPost.ID : key
	) )
};
```